### ISSUE TITLE
all output should be reduced

### ISSUE THREAD

**Issue by bearecinos** [EdiGlacUQ/fenics_ice#32]
If some of the output from the inversion is not needed for further tasks we should write out less variables during [run_inv.py](https://github.com/EdiGlacUQ/fenics_ice/blob/main/runs/run_inv.py#L95-L127) -> each inversion probably makes ~ 2.8 GB of data. This is not reasonable if we will scale up.

The amount of data to write should be left to each user decision by specifying in the `.toml` a list of variables to write. Scripts under runs need to be more general, to avoid copying and pasting your own version of `run_inv.py` as I had to do for the l-curves experiments. 

Logging needs to be reduce! -> is it possible to do less printing of newton iterations? (cc. @jrmaddison )

**Comment by dngoldberg** [comment]
@bearecinos just to quickly check, have you seen if logging can be reduced with inversion.verbose = false in the toml file?

**Comment by bearecinos** [comment]
will add this to the .toml file for smith then! thanks!

**Comment by bearecinos** [comment]
I will add to this issue ... That we need to reduce the output of the whole workflow (mostly for stages after the inversion), if we want to increase the efficiency. I want to make the writing of output **user specific** in every stage. Probably with extra controls in the .toml

for example: 
for run_forward.py ... pass a parameter called: "complete", "reduced", extra reduce output ...

A good example why this is important its in [run_forward.py](https://github.com/EdiGlacUQ/fenics_ice/blob/main/fenics_ice/solver.py#L579)

here we don't pass any other argument to slvr.timestep: 
```# Run the forward model
    Q = slvr.timestep(adjoint_flag=1, qoi_func=qoi_func)
``` 

And in solver.py ... [slvr.timestep()](https://github.com/EdiGlacUQ/fenics_ice/blob/main/fenics_ice/solver.py#L579) we have:

```
def timestep(self, save=1, adjoint_flag=1, qoi_func=None ):
.
.
        if save:
            Hfile = Path(outdir) / "_".join((self.params.io.run_name,
                                             'H_ts.xdmf'))
            Ufile = Path(outdir) / "_".join((self.params.io.run_name,
                                             'U_ts.xdmf'))

            xdmf_hts = XDMFFile(self.mesh.mpi_comm(), str(Hfile))
            xdmf_uts = XDMFFile(self.mesh.mpi_comm(), str(Ufile))

            xdmf_hts.write(H_np, 0.0)
            xdmf_uts.write(U_np, 0.0)
``` 

But if by default save is 1... python will always save this, correctly should be if save == 1 and should be also an else after? saying that if save is something else don't save the output of thickness and ice velocity at every time step, which in our smith_glacier case is every 960 ... having this type of statements with no close loopholes might be tricky

**Comment by bearecinos** [comment]
Dont know if I have reduce enough to close this ... 🤔 I will think about it

**Comment by bearecinos** [comment]
I @dngoldberg this issue is closed now ... I actually managed to add a configuration parameter that decides what type of file to write .. to give the user options, to output in a paraview file style or in xml only. See PR #[51](https://github.com/EdiGlacUQ/fenics_ice/pull/51)

**Comment by dngoldberg** [comment]
thanks @bearecinos. how does your update allow us to change the format within the following:

```
        if save_every_tstep:
            Hfile = Path(outdir) / "_".join((self.params.io.run_name,
                                             'H_ts.xdmf'))
            Ufile = Path(outdir) / "_".join((self.params.io.run_name,
                                             'U_ts.xdmf'))

            xdmf_hts = XDMFFile(self.mesh.mpi_comm(), str(Hfile))
            xdmf_uts = XDMFFile(self.mesh.mpi_comm(), str(Ufile))

            xdmf_hts.write(H_np, 0.0)
            xdmf_uts.write(U_np, 0.0)

**Comment by bearecinos** [comment]
in the toml

[time] 
save_every_tstep: bool = True

Will generate these files but by default is False so no files are ever written. If you need this in a different format we can add a different way of writing them .. but so far I have done nothing with these files as I was working on visualizing the workflow ... I assume that writing this in H5 of xml we can some how visualize them with matplotlib but at the moment there is no code for that...  you can generate the paraview ones ... but again I don't know how to do this in paraview either.

**Comment by dngoldberg** [comment]
yes this was the point of my comment above. it should be handled by the inout functions.. i will open an issue.

**Comment by bearecinos** [comment]
ah sorry I see! yes please do, maybe call it visualize output for every time step ... as is not about reducing output but how to visualize it

**Comment by dngoldberg** [comment]
thanks @bearecinos and as i also said above, this requires "(ii) changing slightly so that the data is written at predefined intervals (or the closest time step to this interval)." 

When i get a chance i will open an issue.

**Comment by bearecinos** [comment]
I am going to re open this ... might be easier ...

**Comment by bearecinos** [comment]
Additional output reduction by PR #68
